src/index.js

```diff
@@ -89,19 +89,19 @@ function visitEntity(entity, entitySchema, bag, options) {
   return id;
 }
 
-function visit(obj, schema, bag, options) {
+function visit(obj, schema, bag, options, collectionKey) {
   if (!isObject(obj) || !isObject(schema)) {
     return obj;
   }
 
   if (schema instanceof EntitySchema) {
-    return visitEntity(obj, schema, bag, options);
+    return visitEntity(...arguments);
```

Comment:
Can we avoid relying on the `arguments` object in these lines and just add `collectionKey` to the original set?